[Ready] *smacks table* Objection!

Urumasi opened the pull request:
## About The Pull Request

Direct extension of #57269

After a [suggestion](https://i.imgur.com/yaeWSnE.png), I decided to make slamming a table while wearing an attorney's badge yell "Objection!!"
Also makes slamming tables a signal

## Why It's Good For The Game

![Objection!](https://i.imgur.com/a2Mny9b.png)

## Changelog
:cl:
add: Lawyers can now object by slamming the table.
code: Table slamming is now handled using signals.
/:cl:

Urumasi committed:
Objection!!

Urumasi commented:
I'm closing this, going to rework it into action buttons

itseasytosee commented:
Don't make it an action button. This PR is perfect. We don't need more screen clutter. There is no need to shove a feature like this in someone's face.

Urumasi committed:
Merge branch 'master' into objection!

Urumasi commented:
Re-opening after trying to find a copyright-free Objection! sound to perhaps play if you're in the Courtroom, but I decided this PR is good enough on it's own

Mothblocks commented on code/game/objects/items/hand_items.dm:
```diff
@@ -151,6 +151,10 @@
 			if(istype(human_user.shoes, /obj/item/clothing/shoes/cowboy))
 				human_user.say(pick("Hot damn!", "Hoo-wee!", "Got-dang!"), spans = list(SPAN_YELL), forced=TRUE)
 				human_user.client?.give_award(/datum/award/achievement/misc/hot_damn, human_user)
+			else if(istype(human_user.w_uniform, /obj/item/clothing/under))
+				var/obj/item/clothing/under/uniform = human_user.w_uniform
+				if(istype(uniform.attached_accessory, /obj/item/clothing/accessory/lawyers_badge))
+					human_user.say("Objection!!", spans = list(SPAN_YELL), forced=TRUE)
```
I think now that we have more than one thing that cares about this, it should just be made into a signal. It's going to be more work for you, but it's cleaner.

Urumasi replied:
I'll look into it

Urumasi committed:
Rework table slam into signals

Urumasi committed:
Linter ate my face, use async calls

Mothblocks commented on code/__DEFINES/dcs/signals.dm:
```diff
@@ -653,6 +653,9 @@
 ///from base of mob/living/simple_animal/hostile/regalrat: (mob/living/simple_animal/hostile/regalrat/king)
 #define COMSIG_RAT_INTERACT "rat_interaction"
 
+///from /obj/item/slapper/attack_obj(obj/O, mob/living/user, params)
```
No `obj/O`--https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Mothblocks commented on code/game/objects/items/hand_items.dm:
```diff
@@ -146,15 +146,7 @@
-				var/obj/item/clothing/under/uniform = human_user.w_uniform
-				if(istype(uniform.attached_accessory, /obj/item/clothing/accessory/lawyers_badge))
-					human_user.say("Objection!!", spans = list(SPAN_YELL), forced=TRUE)
+		SEND_SIGNAL(user, COMSIG_LIVING_TABLE_SLAM, the_table)
```
Your documentation isn't correct--you only send the table, nothing else.

Urumasi committed:
Fix signal doc

Urumasi committed:
Adds the user to signal arguments

Makes it possible to register this signal on a table and get info about who slammed it

Urumasi committed:
Send signal to table for potential future features

Mothblocks commented on code/game/objects/items/hand_items.dm:
```diff
@@ -147,6 +147,7 @@
 			color = COLOR_GREEN
 		user.do_attack_animation(the_table)
 		SEND_SIGNAL(user, COMSIG_LIVING_TABLE_SLAM, user, the_table)
+		SEND_SIGNAL(the_table, COMSIG_LIVING_TABLE_SLAM, user, the_table)
```
This should be a different signal.

Urumasi committed:
Split table slamming signal

81Denton commented:
@Mothblocks Is this good to go?

Mothblocks commented:
yeah, forgot about it sorry
